src/Type/Definition/Type.php

```diff
@@ -55,105 +57,85 @@ abstract class Type implements JsonSerializable
     public $extensionASTNodes;
 
     /**
-     * @return IDType
-     *
      * @api
      */
-    public static function id()
-    {
-        return self::getStandardType(self::ID);
-    }
-
-    /**
-     * @param string $name
-     *
-     * @return (IDType|StringType|FloatType|IntType|BooleanType)[]|IDType|StringType|FloatType|IntType|BooleanType
-     */
-    private static function getStandardType($name = null)
+    public static function id() : ScalarType
     {
-        if (self::$standardTypes === null) {
-            self::$standardTypes = [
-                self::ID      => new IDType(),
-                self::STRING  => new StringType(),
-                self::FLOAT   => new FloatType(),
-                self::INT     => new IntType(),
-                self::BOOLEAN => new BooleanType(),
-            ];
+        if ((static::$standardTypes[self::ID] ?? null) === null) {
+            static::$standardTypes[self::ID] = new IDType();
         }
 
-        return $name ? self::$standardTypes[$name] : self::$standardTypes;
+        return static::$standardTypes[self::ID];
     }
 
     /**
-     * @return StringType
-     *
      * @api
      */
-    public static function string()
+    public static function string() : ScalarType
     {
-        return self::getStandardType(self::STRING);
+        if ((static::$standardTypes[self::STRING] ?? null) === null) {
+            static::$standardTypes[self::STRING] = new StringType();
+        }
+
+        return static::$standardTypes[self::STRING];
     }
 
     /**
-     * @return BooleanType
-     *
      * @api
      */
-    public static function boolean()
+    public static function boolean() : ScalarType
```

Comment:
```suggestion
    public static function boolean() : BooleanType
```

Reply:
Don't think there's anything I can do, as this can also be a `CustomScalarType`...